src/main/kotlin/com/github/g3force/oc/OcExecTask.kt

```diff
@@ -5,57 +5,51 @@ import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.TaskAction
-import org.slf4j.Logger
-import org.slf4j.LoggerFactory
 import java.io.BufferedReader
 import java.io.File
 import java.io.InputStreamReader
 
 abstract class OcExecTask : DefaultTask() {
-
-    private val logger: Logger = LoggerFactory.getLogger(OcExecTask::class.java)
-
-    @Input
-    var args: List<String> = emptyList()
-
     @Input
-    var showOutput: Boolean = false
+    var args: List<String> = ArrayList()
 
     @TaskAction
     open fun doAction() {
-        execute(args)
+        execute(emptyList())
     }
 
     fun execute(arguments: List<String>) {
-        val command: MutableList<String> = mutableListOf(findOcBinary())
+        val config = project.extensions.getByType(OcPluginExtension::class.java)
+        val binary = findOcBinary(config)
+
+        val command: MutableList<String> = mutableListOf(binary)
         command.addAll(arguments)
```

Comment:
It is not clear what's the difference is between `arguments` and `args`. What about renaming `arguments` to `baseCommand`?

Reply:
Agree